### PR TITLE
Edit subscription in controller + tests

### PR DESCRIPTION
**Pull request by aviaowl**

### CODE REVIEW

File: subscription_manager/common/utils.py [hunk 1]
```diff
@@ -119,3 +119,21 @@ def validate_str_field(field: str, none_allowed: bool = False):
...
+    """
+    Validate changes and return updated Subscription object
+    Args:
+        subscription (Subscription): subscription object to change
+        changes (dict): dict contains subscription changes
+    Returns:
+        Subscription: updated Subscription object to send to to Database
+    """
+    updated_subscription = asdict(subscription)
+    for key, value in changes.items():
```
Comment (RomantsovArtur) [inline subscription_manager/common/utils.py:136]: why not updated_subscription.update(changes)?

File: subscription_manager/common/utils.py [hunk 2]
```diff
@@ -119,3 +119,21 @@ def validate_str_field(field: str, none_allowed: bool = False):
...
+) -> Subscription:
+    """
+    Validate changes and return updated Subscription object
+    Args:
+        subscription (Subscription): subscription object to change
+        changes (dict): dict contains subscription changes
+    Returns:
+        Subscription: updated Subscription object to send to to Database
+    """
+    updated_subscription = asdict(subscription)
```
Comment (RomantsovArtur) [inline subscription_manager/common/utils.py:135]: please specify :dict

File: subscription_manager/controller.py [hunk 3]
```diff
@@ -38,21 +38,33 @@ def add_subscription(self, subscription_dict: dict) -> int:
...
              "comment": 'Price has been increased since Jun, 21'}
         Returns:
-            id (int): identifier of changed subscription
+            int: count of changed subscriptions
         """
-        pass
+        try:
+            utils.validate_str_field(field=subscription_name)
+            subscription_to_edit = self.get_subscription_by_name(subscription_name)
+        except SubscriptionException:
```
Comment (RomantsovArtur) [inline subscription_manager/controller.py:58]: At next PR please remove all this unnecessary try, except or add some behaviour if you want to do something in th middle

File: subscription_manager/controller.py [hunk 4]
```diff
@@ -38,21 +38,33 @@ def add_subscription(self, subscription_dict: dict) -> int:
...
+            raise
+        try:
```
Comment (RomantsovArtur) [inline subscription_manager/controller.py:60]: You can combine upper and this try

Comment (aviaowl) [reply subscription_manager/controller.py]: Done

File: subscription_manager/controller.py (hunk 3 above)
Comment (aviaowl) [reply subscription_manager/controller.py]: Done

File: subscription_manager/common/utils.py (hunk 2 above)
Comment (aviaowl) [reply subscription_manager/common/utils.py]: Done

File: subscription_manager/common/utils.py (hunk 1 above)
Comment (aviaowl) [reply subscription_manager/common/utils.py]: Done

**Review by RomantsovArtur — APPROVED** [review]

**Pull request merged**